test/test_unistd.rs

```diff
@@ -390,6 +390,40 @@ fn test_pipe2() {
     assert!(f1.contains(FdFlag::FD_CLOEXEC));
 }
 
+#[test]
+fn test_truncate() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let path = tempdir.path().join("file");
+
+    let mut tmp = File::create(&path).unwrap();
+    const CONTENTS: &[u8] = b"12345678";
+    tmp.write_all(CONTENTS).unwrap();
+    drop(tmp);
```

Comment:
Do you really need to `drop(tmp)`? If so, then I think it's better to simply put the preceding three lines within a `{}` block so `tmp` will lose lexical scope and drop automatically.

Reply:
Not really, but given that we are accessing the file through a different "handle" (filename vs. `File` object), it seems nicer to ensure that the object is closed. Used a scope now.